### PR TITLE
Refactor Dapps API endpoints to follow REST standards - Closes #866 

### LINKED ISSUE #866: Refactor Dapps API endpoints to follow REST standards
Parent Issue: #225 

In order to meet the requirements of the 1.0.0 API, we must refactor the `/api/dapps` endpoints to follow REST standards. - link: https://app.swaggerhub.com/apis/LiskHQ/Lisk/1.0.14

### PR DESCRIPTION
**Pull request by MaciejBaj**
Closes #866
Parent issue: #225

### CODE REVIEW

File: api/http/dapps.js [hunk 1]
```diff
@@ -24,12 +21,9 @@ function DappsHttpApi (dappsModule, app) {
...
 	var router = new Router();
 
-	router.map(dappsModule.internal, {
-		'get /categories': 'categories',
-	});
-
-	router.get('/', httpApi.middleware.sanitize('query', schema.list, dappsModule.internal.list));
-	router.get('/get', httpApi.middleware.sanitize('query', schema.get, dappsModule.internal.get));
+	router.map(dappsModule.shared, {
+		'get /': 'list'
```
Comment (diego-G) [inline api/http/dapps.js]: I suggest to follow same pattern than other modules: `getDapps` instead of `list`

File: api/http/dapps.js [hunk 2]
```diff
@@ -24,12 +21,9 @@ function DappsHttpApi (dappsModule, app) {
```
Comment (diego-G) [inline api/http/dapps.js]: Remove line https://github.com/LiskHQ/lisk/blob/bce120982eec9d6b4271b9d5ae8e090db3f6b650/api/http/dapps.js#L19

File: sql/dapps.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 var DappsSql = {
-	sortFields: ['type', 'name', 'category', 'link'],
+	sortFields: ['name'],
 
 	countByTransactionId: 'SELECT COUNT(*)::int AS "count" FROM dapps WHERE "transactionId" = ${id}',
```
Comment (diego-G) [inline sql/dapps.js]: Please, review the unused functions and remove them. For example, I've checked `search` is obsoleted.
Also check the indentation in the file.

File: api/http/accounts.js [hunk 4]
```diff
@@ -36,7 +36,7 @@ function AccountsHttpApi (accountsModule, app) {
 	});
 
 	if (process.env.TOP && process.env.TOP.toUpperCase() === 'TRUE') {
-		router.get('/top', httpApi.middleware.sanitize('query', schema.top, accountsModule.internal.top));
+		router.get('/top', httpApi.middleware.sanitize('query', schema.top, {}, accountsModule.internal.top));
```
Comment (Isabello) [inline api/http/accounts.js]: Why is this being touched?

File: helpers/httpApi.js [hunk 5]
```diff
@@ -83,16 +83,19 @@ var middleware = {
 	 * Uses req.sanitize for particular endpoint.
 	 * @param {string} property
 	 * @param {Object} schema
+	 * @param {Object} options
 	 * @param {function} cb
 	 * @return {function} Sanitize middleware.
 	 */
-	sanitize: function (property, schema, cb) {
+	sanitize: function (property, schema, options, cb) {
+		//ToDo: Remove optional error codes response handler choice as soon as all modules will be conformed to new REST API standards
```
Comment (Isabello) [inline helpers/httpApi.js]: TODO

Should also have a space before T I believe.

**Pull request merged**